### DISCUSSION TITLE
Merging multiple MAFs

### DISCUSSION THREAD

**Discussion by FredoJones** [PoisonAlien/maftools#778] [Q&A] [answered]
Hello! 
I have multiple .MAF files (one for each sample) and thus cannot read them in using read.maf() function. 
I figured I have to merge them but i was not able to do that using the merge.maf() function (R crashes) or with the cat function in Linux (as the output does not allow to differentiate for the different samples). Do you have any solution to this issue? 
Much appreciated!

**Comment by PoisonAlien** [comment, marked as answer]
Hi,
Is the R crash due to low memory? Could you please post the error message?

**Reply by FredoJones** [reply]
The fatal error is called despite raising working memory to 16x10^6 which is my max. there is no error report, but the whole session crashes. I have tried generating a list with each maf read singularly, then tried to merge with merge_mafs but i get this error: 

mergedmaf <- merge_mafs(mafs = maf.list)
Merging 144 MAF objects
Error in data.table::rbindlist(l = mafs_dat, fill = TRUE, use.names = TRUE) :
Internal error: column 405 of result is determined to be integer64 but maxType=='character' != REALSXP

**Reply by FredoJones** [reply]
maf.list was generated with: 

for (i in mafvector) {
  maf.list[[i]] <- read.maf(maf = i)
}
maf.list contains 144 elements and is 390 MB

**Comment by PoisonAlien** [comment]
Hi,

The issue here seems to be due to differences in column classes (e.g, mix of Integer and character). This is expected when you have many columns, and most of the time you don't need them all. My suggestion is to only select the required columns from the MAF file and work with it. 

Please see issue #722 for a possible solution. Let me know if it helps.
